src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/PartitionedRateLimiter.cs

```diff
@@ -182,48 +214,82 @@ private void ThrowIfDisposed()
             }
         }
 
-        private static void Replenish(DefaultPartitionedRateLimiter<TResource, TKey> limiter)
+        private async Task TimerLoop()
```

Comment:
Nit: at first, I thought we moved the `while (await _timer)` into this method.

```suggestion
        private async Task Heartbeat()
```

I'm not tied to the name. Anything that doesn't make it sound like the whole loop is fine by me.